CampaignAutomation/Fetchers/RLTContentTypeDataModelFetching.swift

```diff
@@ -0,0 +1,31 @@
+//
+//  RLTContentTypeDataModelFetching.swift
+//  RealifeTech-SDK
+//
+//  Created by YOU-HSUAN YU on 2022/5/26.
+//
+
+import Foundation
+
+public protocol RLTContentTypeDataModelFetching {
+    func fetchRLTDataModels(location: String, completion: @escaping (Result<[RLTItem], Error>) -> Void)
+    func fetchRLTContentTypeDataModels(location: String, completion: @escaping (Result<[RLTDataModelConverter.ContentTypeDataModel], Error>) -> Void)
+}
+
+public extension RLTContentTypeDataModelFetching {
+
+    func fetchRLTDataModels(location: String, completion: @escaping (Result<[RLTItem], Error>) -> Void) {
```

Comment:
I can understand that default method implementation can be very convenient, but I would recommend avoiding it. This is a signal of a code smell. There are multiple reasons why you should avoid this approach: 
- default method implementation is similar to inheritance and OOP favours composition over inheritance 
- it is hard to unit test 
- there can be hard to find bugs related to method dispatch 

This is a very good article that summarises it all - [Why You Should Avoid Using Default Implementations in Protocols](https://betterprogramming.pub/swift-why-you-should-avoid-using-default-implementations-in-protocols-eeffddbed46d#:~:text=As%20mentioned%20in%20Swift's%20documentation,can%20also%20add%20new%20ones.)

Reply:
Hi @paleksandrs , thank you for sharing this article!
Sorry that I should mention in the title that this PR is not ready for review as we are still reviewing the architecture.
But we decide not to go for this solution. so I'll close this PR. Thank you!